docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+- [Provider / Router](#provider--router)
+- [Query Container facade](#query-container-facade)
+- [Query Object](#query-object)
+- [Repository](#repository)
+- [Service facade](#service-facade)
+- [Theme](#theme)
+- [Transfer Object](#transfer-object)
+- [Widget](#widget)
+- [Zed Stub](#zed-stub)
+- [Bridge](#bridge)
+- [Plugin](#plugin)
+- [Plugin Interface](#plugin-interface)
+
+# Documentation structure
+
+## Development Use-Cases
+Understanding the development scenarios in which Spryker Commerce Operating System (SCOS) can be utilized is crucial for maximizing its potential. We have outlined specific behaviors and guidelines tailored to different use-cases:
+
+- **Project Development**: If you are developing a project, you will need to adhere to specific project development guidelines to ensure smooth integration.
+- **Boilerplate and Accelerator Contribution**: Contributing to boilerplates or accelerators requires additional considerations. We provide guidelines specific to this use-case to streamline your contributions.
+- **Module Contribution**: For those contributing to modules, there are specialized rules to follow within the module folders. This ensures consistency and compatibility across SCOS.
+
+## Directive Classification
+Throughout this documentation, you will encounter two types of directives:
+- **Convention**: These are mandatory requirements that contributors must adhere to in order to enable specific SCOS features or ensure proper application responses.
+- **Guideline**: While not mandatory, following these guidelines is highly recommended. Doing so promotes long-term code maintainability and facilitates smoother development processes.
+
+# Applications
+Spryker utilises application layers into applications to enable constructing the necessary application architecture for the specific business requirements to provide a quick project start, and long-term maintainability.
+- Backend applications (eg: Zed, Backend API, Backend Gateway, Backoffice, GlueBackend, MerchantPortal) typically use the Zed-, Glue-, Client-, Service-, and Shared application layers.
+- Storefront applications (eg: Yves, Configurator, Glue, GlueStorefront) typically use the Yves-, Glue-, Client-, Service- and Shared application layers.
+
+# Application Layers
+Spryker organises responsibilities/functionalities over a set of application layers (see [Conceptual Overview](https://docs.spryker.com/docs/dg/dev/architecture/conceptual-overview.html)) to enable flexible business logic orchestration across the applications.
+
+The application layers are aggregations of layers (see [Modules and Application Layers](https://docs.spryker.com/docs/dg/dev/architecture/modules-and-application-layers.html))
+- **Zed**: is a **multi-layer composition** of the 4 layers: `Presentation Layer`, `Communication Layer`, `Business Layer`, `Persistence Layer`.
+- **Yves**: is a **single-layered** `Presentation Layer` merged with `Communication Layer`.
```

Comment:
I don't understand why we consider Yves as single-layer?

Reply:
The layers are not distingiushed from each other. You can't find the `Presenttation` or `Communication` directories, like in Zed. I will use another term, i see why the `single-layered` a bit confusing.